tests/monit/test_monit_status.py

```diff
@@ -26,23 +39,69 @@ def disable_lldp(duthosts, enum_rand_one_per_hwsku_frontend_hostname):
-    monit_status_result = duthost.shell("sudo monit status \'lldp|lldpmgrd\'", module_ignore_errors=True)['stdout_lines']
-    indices = [i for i, s in enumerate(monit_status_result) if 'last output' in s]
-    monit_last_output = monit_status_result[indices[0]]
-    if duthost.is_multi_asic:
-        return "/usr/bin/lldpmgrd' is not running in host and in namespace asic0" in monit_last_output
+    """Checks whether alerting message appears as output of command 'monit status' if
+    process `lldpmgrd` was stopped.
+
+    Args:
+        duthost: An AnsibleHost object of DuT.
+
+    Returns:
+        None.
+    """
+    monit_status_result = duthost.shell("sudo monit status 'lldp|lldpmgrd'", module_ignore_errors=True)
+    exit_code = monit_status_result["rc"]
+    pytest_assert(exit_code == 0, "Failed to get Monit status of process 'lldpmgrd'!")
+
+    indices = [i for i, s in enumerate(monit_status_result["stdout_lines"]) if 'last output' in s]
+    if len(indices) > 0:
+        monit_last_output = monit_status_result["stdout_lines"][indices[0]]
+        if duthost.is_multi_asic:
+            return "/usr/bin/lldpmgrd' is not running in host and in namespace asic0" in monit_last_output
+        else:
+            return "/usr/bin/lldpmgrd' is not running in host" in monit_last_output
     else:
-        return "/usr/bin/lldpmgrd' is not running in host" in monit_last_output
+        pytes.fail("Failed to get Monit last output of process 'lldpmgrd'!")
+
 
 def test_monit_status(duthosts, enum_rand_one_per_hwsku_frontend_hostname):
+    """Checks whether the Monit service was running or not.
+
+    Args:
+        duthosts: The fixture returns list of DuTs.
+        enum_rand_one_per_hwsku_frontend_hostname: The fixture randomly picks up
+        a frontend DuT from testbed.
+
+    Returns:
+        None.
+    """
     duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
     monit_status_result = duthost.shell("sudo monit status", module_ignore_errors=True)
 
     exit_code = monit_status_result["rc"]
     pytest_assert(exit_code == 0, "Monit is either not running or not configured correctly")
 
+
 def test_monit_reporting_message(duthosts, enum_rand_one_per_hwsku_frontend_hostname, disable_lldp):
+    """Checks whether the format of alerting message from Monit is correct or not.
+       202012 and newer image version will be skipped for testing since Supervisord
+       replaced Monit to do the monitoring critical processes.
+
+    Args:
+        duthosts: The fixture returns list of DuTs.
+        enum_rand_one_per_hwsku_frontend_hostname: The fixture randomly pick up
+        a frontend DuT from testbed.
+        disable_lldp: The fixture function stops `lldpmgrd` process before testing
+        and restarts `lldpmgrd` process at teardown.
+
+    Returns:
+        None.
+    """
     duthost = duthosts[enum_rand_one_per_hwsku_frontend_hostname]
+
+    pytest_require(parse_version(duthost.kernel_version) <= parse_version("4.9.0"),
```

Comment:
Why don't check ```sonic_version``` with ```duthost.os_vesion```?

Reply:
Good question! Currently Monit is still used to monitor the critical processes in 201911 and older image version. If we used the `duthost.os_version` to differentiate between the 202012 image from 201911 image, I am not sure which `os_version` we can use since new 201911 image may be built in future.

Reply:
I see. Thanks

Reply:
@yozhao101: Why can you simply not check if `duthost.os_vesion` contains `201911`?

Reply:
Oh, we can do that. Great suggestion! Initially what I am thinking is which `os_version` I should use to compare with `duthost.os_version` for example `20191130.73` or `20191130.72`. Yes, we can simplify this logic by just deciding whether `duthost.os_version` contains `201911` or not.

Reply:
Thanks Joe, Bing for great question!